keel-intent/src/main/kotlin/com/netflix/spinnaker/keel/intent/processor/SecurityGroupIntentProcessor.kt

```diff
@@ -69,27 +81,30 @@ class SecurityGroupIntentProcessor
       intent = intent
     ))
 
-    if (currentStateUpToDate(intent.id, currentState, intent.spec)) {
-      return ConvergeResult(listOf(), ConvergeReason.UNCHANGED.reason)
+    if (currentStateUpToDate(intent.id, currentState, intent.spec, changeSummary)) {
+      changeSummary.addMessage(ConvergeReason.UNCHANGED.reason)
+      return ConvergeResult(listOf(), changeSummary)
     }
 
     val missingGroups = missingUpstreamGroups(intent.spec)
     if (missingGroups.isNotEmpty()) {
-      return ConvergeResult(listOf(), "Some upstream security groups are missing: $missingGroups")
+      changeSummary.addMessage("Some upstream security groups are missing: $missingGroups")
+      changeSummary.type = ChangeType.UNHANDLED
```

Comment:
Not sure `UNHANDLED` is the right thing here. Maybe something like... `FAILED_PRECONDITIONS` or something? `UNHANDLED` feels like a short-coming of the intent processor / something not being implemented, which this case is not.

Reply:
`FAILED_PRECONDITIONS` ise much better. Changing.